File: Phase2/src/main/java/PreProcessing.java
package main.java;

import java.io.File;
import java.io.IOException;
import java.net.URL;

public class PreProcessing {
    public static String preprocesses() {
        URL url = Main.class.getResource("../resources/EnglishData");
        if (url == null) return "INVALID PATH!";
        else {
            File directoryPath = new File(url.getPath());
            File[] filesList = directoryPath.listFiles();
            try {
                InvertedIndex index = new InvertedIndex();
                if (filesList == null) return "INVALID FILE LIST!";
                else {
                    addFilesToIndexFiles(filesList, index);
                    new InputScannerView(index);
                    return "end";
                }
            } catch (Exception e) {
                return "an error has happened!";
            }
        }
    }

    private static void addFilesToIndexFiles(File[] filesList, InvertedIndex index) throws IOException {
        for (File file : filesList) {
            index.indexFile(file);
        }
    }
}

File: WebApplication/PreProcessing.cs
using System;
using System.IO;

namespace ConsoleApp1
{
    public class PreProcessing
    {
        static string invalidPathString = "INVALID PATH!";

        public static string Preprocesses(string input)
        {
            string[] files = Directory.GetFiles(@"\..\..\..\Phase-52\TestProject1\");

            try
            {
                IInvertedIndex index = new InvertedIndex();

                addFilesToIndexFiles(files, index);
                var inputScanner = new InputScanner(index);
                var result = inputScanner.GetOrder(input);
                
                var answer = "";
                foreach (var s in result)
                {
                    answer += s;
                    answer += "\n";
                }

                return answer;
            }
            catch (Exception e)
            {
                return "an error has happened!"+e.Message;
            }
        }

        private static void addFilesToIndexFiles(string[] filesList, IInvertedIndex index)
        {
            index.IndexFile(filesList);
        }
    }
}